### PR TITLE
Feature/puppet

### PR DESCRIPTION
**Pull request by frasercobb**
Added puppet support. Let me know of anything to change and i'll complete ASAP.
Sorry about the dirty commit history, still learning git.

Update: 
Cleaned up git history.
Removed copyright.
Cleaned up code after lint tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,6 +10,7 @@
 /test/tmp/
 /test/version_tmp/
 /tmp/
+/.idea/
```
Comment (ianchesal) [inline .gitignore]: What's this for?

**Comment by ianchesal** [comment]
`git rebase -i` is your friend when you're prepping for a PR. Squash all those unnecessary commits.

**Comment by ianchesal** [comment]
There are some Rubocop violations on the code changes. Please run:

    bundle exec rake lint

to check your Ruby syntax against the Rubocop rules for the project.

You should rebase against the `develop` branch to so this merges cleanly. The conflicts are minor and should be pretty obvious and easy to sort out.

Thanks!

File: .gitignore (hunk 1 above)
Comment (frasercobb) [inline .gitignore]: My IDE, RubyMine adds this folder by default.

File: lib/packer/provisioners/puppet/masterless.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Encoding: utf-8
+# Copyright 2015 Fraser Cobb
```
Comment (ianchesal) [inline lib/packer/provisioners/puppet/masterless.rb]: The project license trumps the file comments. I should probably just remove these across all the files in the project...

File: lib/packer/provisioners/puppet/server.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# Encoding: utf-8
+# Copyright 2015 Fraser Cobb
```
Comment (ianchesal) [inline lib/packer/provisioners/puppet/server.rb]: The project license trumps the file comments. I should probably just remove these across all the files in the project...

File: spec/packer/provisioners/puppet/masterless_spec.rb [hunk 4]
```diff
@@ -0,0 +1,155 @@
+# Encoding: utf-8
+# Copyright 2015 Fraser Cobb
```
Comment (ianchesal) [inline spec/packer/provisioners/puppet/masterless_spec.rb]: The project license trumps the file comments. I should probably just remove these across all the files in the project...

File: spec/packer/provisioners/puppet/server_spec.rb [hunk 5]
```diff
@@ -0,0 +1,151 @@
+# Encoding: utf-8
+# Copyright 2015 Fraser Cobb
```
Comment (ianchesal) [inline spec/packer/provisioners/puppet/server_spec.rb]: The project license trumps the file comments. I should probably just remove these across all the files in the project...

**Comment by ianchesal** [comment]
LGTM!

**Pull request merged**